### PR TITLE
[script.video.bgtime_tv] 1.0.8

### PR DESCRIPTION
**Pull request by portokala123**
### Description
Video plugin for bgtime.tv service. Addes support for matrix

### Checklist:
- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [script.foo.bar] 1.0.0

### CODE REVIEW

**Comment by enen92** [comment]
ERROR: Image icon should be explicitly declared in addon.xml <assets>.

File: script.video.bgtime_tv/addon.xml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+		<import addon="script.module.requests" version="2.7.0"/> 
+		<import addon="script.module.simplecache" version="1.0.0" />
+	</requires>
+	<extension point="xbmc.python.script" library="addon.py">
+		<provides>video</provides>
+	</extension>
+	<extension point="xbmc.addon.metadata">
+		<platform>all</platform>
+		<language>en bg</language>
+		<license>The MIT License (MIT)</license>
```
Comment (enen92) [inline script.video.bgtime_tv/addon.xml:14]: please use only "MIT"

File: script.video.bgtime_tv/const.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+from simplecache import  SimpleCache
+import os.path
+
+
+ADDON           = xbmcaddon.Addon()
+VERSION         = ADDON.getAddonInfo('version')
+LANG            = ADDON.getLocalizedString
+ADDON_NAME      = ADDON.getAddonInfo('name')
+ID              = ADDON.getAddonInfo('id')
+PROFILE_PATH    = xbmc.translatePath( ADDON.getAddonInfo('profile') )
```
Comment (enen92) [inline script.video.bgtime_tv/const.py:16]: xbmcvfs.translatePath

File: script.video.bgtime_tv/const.py [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+ADDONPATH       = xbmc.translatePath( ADDON.getAddonInfo('path') )
```
Comment (enen92) [inline script.video.bgtime_tv/const.py:17]: xbmcvfs.translatePath

File: script.video.bgtime_tv/gui.py [hunk 4]
```diff
@@ -0,0 +1,1524 @@
...
+	}
+	data = _json_rpc_request(payload)
+
+	if 'error' in data:
+		try:
+			xbmc.executebuiltin('InstallAddon(inputstream.adaptive)', True)
+			xbmc.executeJSONRPC('{"jsonrpc":"2.0","id":1,"method":"Addons.SetAddonEnabled","params":{"addonid":"inputstream.adaptive","enabled":true}}')
+			return xbmcaddon.Addon('inputstream.adaptive')
+		except:
+			xbmcgui.Dialog().ok('Missing inputstream.adaptive add-on', 'inputstream.adaptive add-on not found or not enabled.This add-on is required to view DRM protected content.')
```
Comment (enen92) [inline script.video.bgtime_tv/gui.py:73]: this needs to be localized as it is shown in the ui

File: script.video.bgtime_tv/gui.py [hunk 5]
```diff
@@ -0,0 +1,1524 @@
...
+		token = self.getToken();
+
+		if cached is not None and is_menu is None:
+			return cached
+		else:
+			url = const.SITE_PATH + url
+			token = self.getToken();
+
+			if not token:
+				ans = dialog.yesno(const.LANG(32003), 'За да използвате услугата трябва да се впишете')
```
Comment (enen92) [inline script.video.bgtime_tv/gui.py:371]: localize

File: script.video.bgtime_tv/gui.py [hunk 6]
```diff
@@ -0,0 +1,1524 @@
...
+				"device":xbmc.getInfoLabel("System.KernelVersion"),
+				"display":xbmc.getInfoLabel("System.FriendlyName"),
+				"model":xbmc.getInfoLabel("System.BuildDate"),
+				"product":"",
+				"push_id":"",
+				"uuid":""
+			})
+		}
+
+	def writeInFile(self):
```
Comment (enen92) [inline script.video.bgtime_tv/gui.py:525]: with open(const.TOKEN_FILEPATH, "w+") as fopen:
    f.write()

File: script.video.bgtime_tv/gui.py [hunk 7]
```diff
@@ -0,0 +1,1524 @@
...
+		fopen = open(const.TOKEN_FILEPATH, "w+")
+		fopen.write(self.token + " " +  const.ADDON.getSetting('username'))
+		fopen.close()
+
+	def openReadFile(self):
+
+		if os.path.isfile(const.TOKEN_FILEPATH):
+			fopen = open(const.TOKEN_FILEPATH, "r")
```
Comment (enen92) [inline script.video.bgtime_tv/gui.py:533]: same

**Review by enen92 — CHANGES_REQUESTED** [review]
see comments

**Comment by portokala123** [comment]
All problems fixed. I also found some bugs in navigaition, so i fixed them also.

**Review by enen92 — APPROVED** [review]

**Comment by enen92** [comment]
Please check Travis failure:
ERROR: Image icon should be explicitly declared in addon.xml <assets>.

**Comment by enen92** [comment]
Info: https://kodi.wiki/view/Addon.xml#.3Cassets.3E

**Review by enen92 — APPROVED** [review]

**Pull request merged**